commit 30325e23ba0d40567cc4ef78e4ba0c3776ef0c06
Author: Jeff King <peff@peff.net>
Date:   Wed Jun 4 16:55:52 2025 -0400

    curl: fix integer variable typechecks with curl_easy_setopt()
    
    As discussed in the previous commit, we should be passing long integers,
    not regular ones, to curl_easy_setopt(), and compiling against curl 8.14
    loudly complains if we don't.
    
    That patch fixed integer constants by adding an "L". This one deals with
    actual variables.
    
    Arguably these variables could just be declared as "long" in the first
    place. But it's actually kind of awkward due to other code which uses
    them:
    
      - port is conceptually a short, and we even call htons() on it (though
        weirdly it is defined as a regular int).
    
      - ssl_verify is conceptually a bool, and we assign to it from
        git_config_bool().
    
    So I think we could probably switch these out for longs without hurting
    anything, but it just feels a bit weird. Doubly so because if you don't
    set USE_CURL_FOR_IMAP_SEND set, then the current types are fine!
    
    So let's just cast these to longs in the curl calls, which makes what's
    going on obvious. There aren't that many spots to modify (and as you can
    see from the context, we already have some similar casts).
    
    Signed-off-by: Jeff King <peff@peff.net>
    Signed-off-by: Junio C Hamano <gitster@pobox.com>

diff --git a/git-core/imap-send.c b/git-core/imap-send.c
index 27dc033c7f..2e812f5a6e 100644
--- a/git-core/imap-send.c
+++ b/git-core/imap-send.c
@@ -1420,7 +1420,7 @@ static CURL *setup_curl(struct imap_server_conf *srvc, struct credential *cred)
 
 	curl_easy_setopt(curl, CURLOPT_URL, path.buf);
 	strbuf_release(&path);
-	curl_easy_setopt(curl, CURLOPT_PORT, srvc->port);
+	curl_easy_setopt(curl, CURLOPT_PORT, (long)srvc->port);
 
 	if (srvc->auth_method) {
 		struct strbuf auth = STRBUF_INIT;
@@ -1433,8 +1433,8 @@ static CURL *setup_curl(struct imap_server_conf *srvc, struct credential *cred)
 	if (!srvc->use_ssl)
 		curl_easy_setopt(curl, CURLOPT_USE_SSL, (long)CURLUSESSL_TRY);
 
-	curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, srvc->ssl_verify);
-	curl_easy_setopt(curl, CURLOPT_SSL_VERIFYHOST, srvc->ssl_verify);
+	curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, (long)srvc->ssl_verify);
+	curl_easy_setopt(curl, CURLOPT_SSL_VERIFYHOST, (long)srvc->ssl_verify);
 
 	curl_easy_setopt(curl, CURLOPT_READFUNCTION, fread_buffer);
 
